misc/python/materialize/mzbuild.py

```diff
@@ -605,31 +614,27 @@ def build(self) -> None:
         ]
         spawn.runv(cmd, stdin=f, stdout=sys.stderr.buffer)
 
-    def acquire(self) -> None:
-        """Download or build the image if it does not exist locally."""
-        if self.acquired:
-            return
-
-        ui.header(f"Acquiring {self.spec()}")
-        try:
-            spawn.runv(
-                ["docker", "pull", self.spec()],
-                stdout=sys.stderr.buffer,
-            )
-        except subprocess.CalledProcessError:
-            self.build()
-        self.acquired = True
-
-    def ensure(self) -> None:
-        """Ensure the image exists on Docker Hub if it is publishable.
+    def acquire(self) -> bool:
+        """Download the image if it does not exist locally. Returns whether it was found."""
+        if not self.acquired:
+            ui.header(f"Acquiring {self.spec()}")
+            try:
+                spawn.runv(
+                    ["docker", "pull", self.spec()],
+                    stdout=sys.stderr.buffer,
+                )
+            except subprocess.CalledProcessError:
+                pass
+            else:
+                self.acquired = True
+        return self.acquired
```

Comment:
`DependencySet.acquire` and `ResolvedImage.acquire` should have the same behavior as they share the same name. Recommend renaming this method to `pull` to reflect its new behavior.

I also think you can ditch the `acquired` flag, because this function will no longer be called multiple times per `ResolvedImage`.

```suggestion
    def try_pull(self) -> bool:
        """Pull the image if it does not exist locally. Returns whether it was found."""
        ui.header(f"Trying to pull {self.spec()}")
        try:
            spawn.runv(
                ["docker", "pull", self.spec()],
                stdout=sys.stderr.buffer,
            )
            return True
        except subprocess.CalledProcessError:
            return False
```

Reply:
It won't be called multiple times, but acquired is still set in the __init__ method for locally known images. So I'll keep it.